### PR TITLE
feat(esl-popup): container attribute to set up bounds of visibility

### PR DESCRIPTION
**Pull request by dshovchko**
In scope of this PR:
- added container attribute to set up bounds of visibility (esl-popup, esl-note, esl-note-group)
- refactored code to avoid duplication
- updated expamples

### CODE REVIEW

File: src/modules/esl-footnotes/core/esl-note-group.ts [hunk 1]
```diff
@@ -4,17 +4,25 @@ import {ready} from '../../esl-utils/decorators/ready';
 
 import type {ESLNote} from './esl-note';
 
+interface Attribute {
+  name: string;
+  value: string;
+}
```
Comment (ala-n) [inline src/modules/esl-footnotes/core/esl-note-group.ts:10]: Let's discuss changes in esl-note-group.ts today
Attribute propagation logic looks a little bit overcomplicated.

**Review by ala-n — COMMENTED** [review]
Have no concerns except esl-note-group.ts changes

File: src/modules/esl-footnotes/core/esl-note-group.ts (hunk 1 above)
Comment (dshovchko) [reply src/modules/esl-footnotes/core/esl-note-group.ts]: You are right! It looks like an elephant on the bike. :)
I have already reworked esl-note-group.
What do you think about the new variant?

**Review by ala-n — APPROVED** [review]

**Review by abarmina — APPROVED** [review]

**Review by fshovchko — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

**Pull request merged**